src/main/java/picard/sam/markduplicates/util/ReadEndsForMarkDuplicatesCodec.java

```diff
@@ -54,12 +54,13 @@ public void encode(final ReadEndsForMarkDuplicates read) {
             this.out.writeShort(read.score);
             this.out.writeShort(read.libraryId);
             this.out.writeByte(read.orientation);
+            this.out.writeByte(read.read2CoordinateRequiresSerialize ? 1 : 0);
```

Comment:
I'm not totally sure you really want to serialize this every time.  It might make more sense to add an optional constructor parameter to the codec and keep track of it that way.  I haven't tried to thread it through everywhere to see how awful that is though, so maybe this is the best way.